### ISSUE TITLE
Can't use single quote in identifiers

### ISSUE THREAD

**Issue by darkf** [puffnfresh/roy#57]
Tested at [the website](http://roy.brianmckenna.org/), the following won't tokenize:

```let a' = 5

```
```

**Comment by gregwebs** [comment]
I don't think this helps Haskell much. a2 or a_ works fine, but usually the best is to use more descriptive names.

**Comment by darkf** [comment]
a2 or a_ is ugly, and a' comes from some mathematics. I don't see any reason to leave this out, it's just a small change to the lexer.

**Comment by puffnfresh** [comment]
@darkf sadly it's not just a small change.

We use the `unicode-categories` package to give us ECMAScript 5.1 identifiers. The code goes like this:

```
var IDENTIFIER = new RegExp(
    unicode.ECMA.identifier.source.replace('\\u03BB', '')
);
```

It's definitely possible to do another replace on the regex:

```
var IDENTIFIER = new RegExp(
    unicode.ECMA.identifier.source.replace('\\u03BB', '').replace(/]([^\]]*)$/, "']$1")
);
```

But that's very hacky :-1:
